DEVELOPMENT.md

```diff
@@ -0,0 +1,162 @@
+
+# Ontotext GraphQL Playground Development Notes
+
+The **Ontotext GraphQL Playground** is a project designed to provide an intuitive interface for exploring and interacting with GraphQL APIs. It builds on a forked repository of [GraphiQL](https://github.com/graphql/graphiql) and enhances it with additional functionality by integrating it into a reusable StencilJS-based web component. This setup ensures flexibility, reusability, and compatibility with other Ontotext projects, starting with the GraphDB Workbench.
+
+## Project Structure
+
+The project is organized as follows:
+
+- **`graphiql/` (Submodule)**  
+  A [forked version](https://github.com/Ontotext-AD/graphiql-fork) of the popular [GraphiQL](https://github.com/graphql/graphiql) library. This submodule contains the core GraphQL playground functionality and acts as the foundation for this project.
+
+- **`ontotext-graphql-playground-component/`**  
+  A [StencilJS](https://stenciljs.com/) web component that wraps the `graphiql` library and publishes it to npm for use in other projects. This component provides a framework-agnostic way to embed the GraphQL Playground in various applications.
+
+## Workflow
+
+The project development workflow follows these steps:
+
+### 1. **Setting up the Repository**
+- Clone the repository, including the submodule:
+  ```bash
+  git clone --recurse-submodules https://github.com/Ontotext-AD/ontotext-graphql-playground
+  ```
+  > **Note**: You need an SSH public key registered with your Git account and sufficient permissions to access the submodule repository.
+
+- If the submodule is not initialized, run:
+  ```bash
+  git submodule update --init --recursive
+  ```
+
+### 2. **Working with the `graphiql` Submodule**
+- The `graphiql` submodule resides in its own folder and can be updated or customized as needed.
+- Make changes in the [forked repository](https://github.com/Ontotext-AD/graphiql-fork) to introduce Ontotext-specific features.
+- Ensure compatibility with the StencilJS component by maintaining API stability.
+
+### 3. **Developing the StencilJS Component**
+Check out stenciljs docs [here](https://stenciljs.com/docs/my-first-component).
+- Navigate to the `ontotext-graphql-playground-component/` directory.
+- Install dependencies:
+  ```bash
+  npm install
+  ```
+- Start the development server:
+  ```bash
+  npm run start
+  ```
+- Update the StencilJS component to wrap `graphiql` features with Ontotext-specific styling and behavior.
+
+### 4. **Building and Publishing**
```

Comment:
This is expected to be automated in jenkins. Right?

Reply:
yes